### ISSUE TITLE
Ensure replayed ladder battles display simulation as it happened

### ISSUE THREAD

**Issue by hamptonsmith** [codecombat/codecombat#651]
Currently, when replaying a ladder battle (via the "View your victory" link, e.g.), the battle is simulated anew using the latest version of the players' code and any level and unit parameters.  This can and does yield an entirely different battle, sometimes changing the outcome as well.

Ideally, the player would see the battle as it happened, either because all parameters, including the code, are carefully snapshotted at that moment (directly to GitHub?  How cool would that be?), or because the simulation is recorded and played back.

This is both more fun (see historical battles!) and more useful (I can carefully study why I lost, even if my opponent changes and breaks their code in the mean-time.)

**Comment by sotonin** [comment]
Yes. consistency definately needs some work. I don't feel confident that what I see is what the system calculated. (losses that when clicked on show the opponent with broken code x under them)
